hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java

```diff
@@ -245,6 +247,12 @@ public DatanodeStateMachine(DatanodeDetails datanodeDetails,
     queueMetrics = DatanodeQueueMetrics.create(this);
   }
 
+  @VisibleForTesting
+  public DatanodeStateMachine(DatanodeDetails datanodeDetails,
```

Comment:
I'm not a fan of it either but feel guilty repeating `null, null, null, null.....`  10 times in tests. It could be easy to create such a constructor in a test folder, but that would not be shared across maven components.